tools/gluc/sources/main.cpp

```diff
@@ -1,5 +1,31 @@
-#include "IRGen/IRGen.hpp"
+#include "Scanner.hpp"
 
-int main() {
-    IRGen irgen;
+#include "llvm/Support/ErrorOr.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/raw_ostream.h"
+
+int main(int argc, char *argv[])
+{
+    if (argc < 2) {
+        llvm::errs() << "Usage: " << argv[0] << " <filename>" << '\n';
+        return 1;
+    }
+
+    llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> bufferOrErr
+        = llvm::MemoryBuffer::getFile(argv[1]);
+
+    if (!bufferOrErr) {
+        llvm::errs() << "Error reading file: "
+                     << bufferOrErr.getError().message() << '\n';
+        return 1;
+    }
+
+    glu::Scanner scanner((*bufferOrErr).get());
```

Comment:
is there no `->` operator?
```suggestion
    glu::Scanner scanner(bufferOrErr->get());
```